main.js

```diff
@@ -0,0 +1,226 @@
+    return website;
+    }
+}
+
+// function handles showing all of the markers
+// on the map when an address is not specified
+// uses async/await method to fetch json
+async function displayAllMarkers(map) {
+  console.log('were gonna display all the markers');
+  // create an array to hold to lats and longs of all restaurants
+  let latsLongs = [];
+
+  // fetch all restaurant names & get json, wait to finish
+  let response = await fetch('all.json');
+  let json = await response.json();
+  let address;
+  let oneLatLong;
+  let website;
+
+  // find lat and long for each restaurant and push to array
+  for await (const name of json) {
+    address = await getAddress(name);
+    oneLatLong = await getLatLon(address);
+    website = await getWebsite(name);  // get website
+    oneLatLong.website = website;      // add website to object 
+    latsLongs.push(oneLatLong);
+  }
+
+  // for each lat and long, add a marker to the map
+  latsLongs.forEach(latlon => {
+    L.marker([latlon.lat, latlon.lon]).addTo(map)
+    // set click event to go to restaurant's website
+    .on('click', () => document.location.href = latlon.website);
+  })
+}
+
+// takes an address as an argument
+async function getLatLon(address) {
+  // fetch json data for address
+  const response = await fetch(`https://nominatim.openstreetmap.org/search/?q=${address}&format=json`);
+  const json = await response.json();
+  // return the lat and lon as an object
+  return {lat: json[0].lat, lon: json[0].lon};
+}
+
+// function creates the links for all restaurant names in all.json
+function createLinks() {
+  const header = document.getElementById('header');
+  const headerLink = document.createElement('a');
+  headerLink.textContent = 'Yelpington!';
+  headerLink.href = 'index.html';
+  header.appendChild(headerLink);
+
+  // get ul element that will hold the links
+  const linkContainer = document.getElementById('links');
+  // fetch all addresses, convert to json
+  fetch('all.json').then(response => response.json())
+  .then(json => {
+    // for each address in json
+    json.forEach(element => {
+
+      // create list item and anchor elements 
+      const listItem = document.createElement('li');
+      const link = document.createElement('a');
+
+      //set text content the the name of the restaurant. should capitalize
+      link.textContent = spaceRestaurantName(element);
+      // set path to query the restaurant name
+      link.href = `index.html?restaurant=${element}`;
+
+      // append link to li and li to ul 
+      listItem.appendChild(link);
+      linkContainer.appendChild(listItem);
+    });
+  })
+}
+
+// replaces dashes in a restaurant name with spaces, capitalize in here?
+function spaceRestaurantName(restaurant) {
+  restaurant = restaurant.split('-');
+  console.log({restaurant});
+  // capitalize words in restaurant name
+  restaurant = restaurant.map(word => {
+    console.log('word: ' + word[0].toUpperCase() + word.slice(1));
```

Comment:
You should do `word.slice(1).toLowerCase()` to make sure the rest of the word doesn't have any weird capitalization going on.